hail/src/main/scala/is/hail/rvd/RVDPartitioner.scala

```diff
@@ -127,6 +133,16 @@ class RVDPartitioner(
     RVDPartitioner.generate(newKType, rangeBounds)
   }
 
+  def selectKey(newKType: TStruct): RVDPartitioner = {
+    require(newKType.isConstructibleFrom(kType))
+    val newRangeBounds = rangeBounds.map { (interval: Interval) =>
+      interval.copy(
+        start = new SelectFieldsRow(interval.start, kType, newKType),
+        end = new SelectFieldsRow(interval.start, kType, newKType))
+    }
```

Comment:
This isn't safe. The range bounds of a partitioner aren't structs of type `kType`, they're prefixes of the fields of `kType`. If `newKType` is a subset/permutation, then the projection may not be a prefix of `newKType`. This is why the coerce logic makes use of `nPreservedFields`, it's the largest prefix which is preserved under the map from old key to new key. The `coarsen` operation is the only version of subsetting the key that is generically valid for any partitioner.

Reply:
I've completely removed these changes once I realized that the behavior in the original PR is exactly the behavior I need.